src/Cli/dotnet/Program.cs

```diff
@@ -182,11 +179,18 @@ internal static int ProcessArgs(string[] args, TimeSpan startupTime, ITelemetry
                         isDotnetBeingInvokedFromNativeInstaller = true;
                     }
 
+                    // Override the workload integrity check if the command is being ran in CI.
+                    if (new CIEnvironmentDetectorForTelemetry().IsCIEnvironment())
```

Comment:
I've set it so that the environment variable (if present) is honored first. If the environment variable is not present, the default value is set to `true` (meaning, skip the check) if running in CI. Otherwise, false.